src/Elastic.Apm/OpenTelemetry/ElasticActivityListener.cs

```diff
@@ -0,0 +1,173 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+#if NET5_0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticListeners;
+using Elastic.Apm.DistributedTracing;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+
+namespace Elastic.Apm.OpenTelemetry
+{
+	public class ElasticActivityListener
+	{
+		internal readonly ConcurrentDictionary<string, Span> ActiveSpans = new();
+		internal readonly ConcurrentDictionary<string, Transaction> ActiveTransactions = new();
+
+		internal ElasticActivityListener(IApmAgent agent, Tracer tracerInternal)
+		{
+			_logger = agent.Logger?.Scoped(nameof(ElasticActivityListener));
+			_tracer = tracerInternal;
+			listener = new ActivityListener
+			{
+				ActivityStarted = ActivityStarted,
+				ActivityStopped = ActivityStopped,
+				ShouldListenTo = _ => true,
+				Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData
+			};
+			ActivitySource.AddActivityListener(listener);
+		}
+
+		private readonly IApmLogger _logger;
+		internal Tracer _tracer;
+
+
+		private Action<Activity> ActivityStarted =>
+			activity =>
+			{
+				_logger.Trace()?.Log($"ActivityStarted: name:{activity.DisplayName} id:{activity.Id} traceId:{activity.TraceId}");
+
+				if (KnownListeners.KnownListenersList.Contains(activity.DisplayName))
```

Comment:
It looks like this is used to not duplicate transactions/spans for Activities for which there is already a DiagnosticListener. For those Activity backed by an ActivitySource, would implementing `listener.ShouldListenTo` perform better?

Should this filtering be extensible to take into account other Elastic APM integrations that are backed by activities, like Azure Service Bus, Storage, MongoDB, and other custom ones a user may have implemented?